### ISSUE TITLE
Documented example doesn't seem to work

### ISSUE THREAD

**Issue by lee-b** [dhruvrajvanshi/crystal-parsec#3]
I'm VERY new to crystal, so forgive me if wrong, but the documented example doesn't seem to work:

Compiling:

```
require "./docsrc/*"

require "parsec"
include Parsec

# TODO: Write documentation for `Docsrc`
module Docsrc

  p = char 'c'
  p.parse("c") # => 'c'

end
```

gives:

```
Error in src/docsrc.cr:9: instantiating 'char(Char)'

  p = char 'c'
      ^~~~

in lib/parsec/src/parsec.cr:91: instantiating 'Parsec::Parser(A):Class#new()'

    Parser.new do |state|
           ^~~

in lib/parsec/src/parsec.cr:36: instance variable '@block' of Parsec::Parser(Char) must be Proc(Parsec::ParseState, Parsec::ParseResult(Char)), not Proc(Parsec::ParseState, (Parsec::ParseResult::Error(Char) | Parsec::ParseResult::Success(Char)))

      @block = block
      ^~~~~~
```

**Comment by dhruvrajvanshi** [comment]
I can't seem to replicate this with the latest version of crystal.
Can you tell me your crystal version and the version of parsec and crz that you're using?
It will be in your lib/crz/shard.yml and lib/parsec/shard.yml files.

**Comment by lee-b** [comment]
I just reproduced it with a completely new crystal project directory:

```
$ grep version lib/*/shard.yml
lib/crz/shard.yml:version: 0.2.0
lib/parsec/shard.yml:version: 0.1.0
lib/parsec/shard.yml:    version: ~> 0.2.0
```

Minimal project directory zip, [parsec_test.zip](https://github.com/dhruvrajvanshi/crystal-parsec/files/2315881/parsec_test.zip), attached.

**Comment by lee-b** [comment]
```
$ crystal --version
Crystal 0.25.1 [b782738ff] (2018-06-27)

LLVM: 4.0.0
Default target: x86_64-unknown-linux-gnu
```

**Comment by dhruvrajvanshi** [comment]
This is an issue with crystal version 0.25. I suspect it has something to do with [this issue](https://github.com/crystal-lang/crystal/pull/6351). It has been fixed in 0.26. You can upgrade your crystal compiler to the latest version.

**Comment by lee-b** [comment]
Ahh, thanks.  Sorry to trouble you in that case.  Thought I was running the latest version.
